### ISSUE TITLE
error NETSDK1064: Package DnsClient, version 1.2.0 was not found

### ISSUE THREAD

**Issue by bbehling** [microsoft/containerregistry#20]
When building a docker image and pulling mcr.microsoft.com/dotnet/core/sdk:3.1  an error is thrown. Note this is also breaking on Elastic Beanstalk  when trying to deploy a new version. This build was working two months ago.

usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1064: Package DnsClient, version 1.2.0 was not found. It might have been deleted since NuGet restore. Otherwise, NuGet restore might have only partially completed, which might have been due to maximum path length restrictions. [/app/myApp.csproj]
The command '/bin/sh -c dotnet publish -c Release -o out' returned a non-zero code: 1

dockerfile -

FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS build-env
WORKDIR /app

COPY *.csproj ./
RUN dotnet restore

COPY . ./
RUN dotnet publish -c Release -o out

FROM mcr.microsoft.com/dotnet/core/aspnet:3.1
WORKDIR /app
COPY --from=build-env /app/out .
EXPOSE 5000
ENTRYPOINT ["dotnet", "myApp.dll"]

Error Log from AWS EB -  

Application deployment failed at 2020-04-12T04:58:20Z with exit status 1 and error: Hook /opt/elasticbeanstalk/hooks/appdeploy/pre/03build.sh failed.

cat: Dockerrun.aws.json: No such file or directory
cat: Dockerrun.aws.json: No such file or directory
cat: Dockerrun.aws.json: No such file or directory
3.1: Pulling from dotnet/core/sdk
Digest: sha256:cad1f51e65af52e5551e5182b163326c8fceca93d0dbb9e8de8d85b95748e4e2
Status: Image is up to date for mcr.microsoft.com/dotnet/core/sdk:3.1
Successfully pulled mcr.microsoft.com/dotnet/core/sdk:3.1
Sending build context to Docker daemon 9.972MB
Step 1/11 : FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS build-env
---> 597ba2726813
Step 2/11 : WORKDIR /app
---> Using cache
---> 68651d922492
Step 3/11 : COPY *.csproj ./
---> Using cache
---> 114b6f8c3b96
Step 4/11 : RUN dotnet restore
---> Using cache
---> 23d6f511441d
Step 5/11 : COPY . ./
---> Using cache
---> 5ed579d49642
Step 6/11 : RUN dotnet publish -c Release -o out
---> Running in 4434565a6968
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

Restore completed in 156.69 ms for /app/BacktraderDataApi.csproj.
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1064: Package DnsClient, version 1.2.0 was not found. It might have been deleted since NuGet restore. Otherwise, NuGet restore might have only partially completed, which might have been due to maximum path length restrictions. [/app/BacktraderDataApi.csproj]
The command '/bin/sh -c dotnet publish -c Release -o out' returned a non-zero code: 1
Failed to build Docker image aws_beanstalk/staging-app, retrying...
Sending build context to Docker daemon 9.972MB
Step 1/11 : FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS build-env
---> 597ba2726813
Step 2/11 : WORKDIR /app
---> Using cache
---> 68651d922492
Step 3/11 : COPY *.csproj ./
---> Using cache
---> 114b6f8c3b96
Step 4/11 : RUN dotnet restore
---> Using cache
---> 23d6f511441d
Step 5/11 : COPY . ./
---> Using cache
---> 5ed579d49642
Step 6/11 : RUN dotnet publish -c Release -o out
---> Running in 4c6139bea64b
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

Restore completed in 109 ms for /app/BacktraderDataApi.csproj.
/usr/share/dotnet/sdk/3.1.201/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1064: Package DnsClient, version 1.2.0 was not found. It might have been deleted since NuGet restore. Otherwise, NuGet restore might have only partially completed, which might have been due to maximum path length restrictions. [/app/BacktraderDataApi.csproj]
The command '/bin/sh -c dotnet publish -c Release -o out' returned a non-zero code: 1
Failed to build Docker image aws_beanstalk/staging-app: ximum path length restrictions. [/app/BacktraderDataApi.csproj]
The command '/bin/sh -c dotnet publish -c Release -o out' returned a non-zero code: 1. Check snapshot logs for details.

**Comment by toddysm** [comment]
@bbehling Looking at the output, this doesn't seem to be a problem with the image but it looks to me more of an issue with the dependencies you have in the *.csproj file. The way I understand the error is that the package is not anymore in the local folder (i.e. it is not installed) and this may be the result of the restore operation.

Posting the project file may be helpful.

cc: @MichaelSimons who can provide more insights

**Comment by MichaelSimons** [comment]
@bbehling - I see the [package](https://www.nuget.org/packages/DnsClient/1.2.0) does exist on NuGet.  Is this something that worked in the past and now just started failing or is this a new Dockerfile you are trying to build?

This does not appear to be related to MCR so I would recommend closing this particular issue.  This looks like a question well-suited for [StackOverflow](https://stackoverflow.com/).  The StackOverflow audience is much larger and you will have better luck getting a good answer there.

**Comment by bbehling** [comment]
Hi.

I did post this on github and another user commented he was having the same problem. 

Yes, the restore command was working February 1st when I published this API to AWS EB.

I narrowed the problem down to running
RUN dotnet restore

If I do not restore, the docker image will build. Note, that outside of docker, I can run the restore command which works.

Here is the content of my .csproj file. There isn't a dependency to the DnsClient library. However, maybe the MongoDB driver requires it because the DnsClient.dll  exists in the release directory after building.

<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="MongoDB.Driver" Version="2.10.1" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
  </ItemGroup>


</Project>

Thank you

**Comment by toddysm** [comment]
@bbehling I don't think the .cproj file went through. Also, can you link to the other issue you mentioned above?

**Comment by toddysm** [comment]
@bbehling As @MichaelSimons suggested, this may be a question for StackOverflow. Because this is not related to problems with Microsoft Container Registry, I will close the issue by the end of the week. Please let me know if you have any concerns with that.

**Comment by bbehling** [comment]
Here is the csproj file.

<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="MongoDB.Driver" Version="2.10.1" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
  </ItemGroup>


</Project>

**Comment by bbehling** [comment]
This is the issue I opened on stackoverflow.
https://stackoverflow.com/questions/61167032/error-netsdk1064-package-dnsclient-1-2-0-was-not-found

A few other people are having issues with the Microsoft Container Register. Since others are having similar issues, seems it's possible this could be an issue with MCR.

**Comment by toddysm** [comment]
@bbehling MCR only serves container images. The problem you are facing is that the dotnet build command which is run while you are building your image fails because the the DnsClient is not available. The DnsClient is fetched from NuGet and not MCR.

The StackOverflow post has a suggestion to use the `--no-cache` option, which you may want to try. 

However, looking deeper into your error above, I would also suggest digging more into following:
**Failed to build Docker image aws_beanstalk/staging-app: ximum path length restrictions.**
As the error suggests, the issue may be because the path is too long and that is why the build fails. You may need to do `ls` for all the folders recursively to troubleshoot the path lengths.

Also, make sure the project builds properly outside an image before building the image.

Once again, because this doesn't seem to be an issue with pulling images from MCR, we will proceed with closing it by the end of the week.

**Comment by bbehling** [comment]
@toddysm One of the troubleshooting steps I did take was to build the application outside of Docker which built successfully.

Also, as I posted, this error maximum path restrictions does not just occur in AWS Beanstalk. This is the error thrown when building the Docker image locally as well.

So if this is not a Docker or MCR related issue, what is the explanation and cause of this issue? The Docker file was copied from [Microsoft’s own tutorial](https://docs.docker.com/engine/examples/dotnetcore/), and failed to build between the Dates of this post, and apparently according to the another stackoverflow post, up to April 29th.
